Reorganize the feature matrix

ptgott opened the pull request:
This change organizes the feature matrix according to the latest product categorization scheme, and moves the feature matrix into its own guide to make it more discoverable. While there is opportunity to rework the rows within the feature matrix, this change minimizes the scope of the first iteration by reorganizing the existing rows while leaving them for the most part unchanged.

In the new feature matrix page, each Teleport product has its own H2-level section. Product descriptions come from official Marketing language, focusing on concrete statements about what each product comprises and how it works and leaving out statements of value (which are not typically found in documentation). This change does not include sections for categories with no rows in the current table.

ptgott committed:
Reorganize the feature matrix

This change organizes the feature matrix according to the latest product
categorization scheme, and moves the feature matrix into its own guide
to make it more discoverable. While there is opportunity to rework the
rows within the feature matrix, this change minimizes the scope of the
first iteration by reorganizing the existing rows while leaving them for
the most part unchanged.

In the new feature matrix page, each Teleport product has its own
H2-level section. Product descriptions come from official Marketing
language, focusing on concrete statements about what each product
comprises and how it works and leaving out statements of value (which
are not typically found in documentation). This change does not include
sections for categories with no rows in the current table.

zmb3 commented on docs/pages/teleport-editions.mdx:
```diff
@@ -0,0 +1,115 @@
+
+## Teleport Zero Trust Access
+
+Teleport Zero Trust Access provides engineers with just-in-time,
```
The _just-in-time_ part refers to Access Requests (I assume), which is part of Teleport Identity Governance, not part of Zero Trust Access.

ptgott replied:
I copied this paragraph from the Zero Trust Access messaging guide ([Google Drive](https://docs.google.com/document/d/15yNRcMQJuwIeeoouzF1CeVwg8ifcCACj76XWB0CN7Pc/edit?tab=t.0)), but I agree that it's incorrect. I'll fix.

zmb3 commented on docs/pages/teleport-editions.mdx:
```diff
@@ -0,0 +1,115 @@
+## Teleport Zero Trust Access
+
+Teleport Zero Trust Access provides engineers with just-in-time,
+least-privileged access to applications, servers, databases, Kubernetes
```
Is there any reason why we're leaving out desktops?

ptgott replied:
I'm not sure—I copied this from the messaging guide to ensure that this is consistent with our official language. I'll add a mention of desktops.

zmb3 commented on docs/pages/teleport-editions.mdx:
```diff
@@ -0,0 +1,115 @@
+management and access control, improving infrastructure resiliency by securing
+system and data access between machines and workloads. 
+
+Machine & Workload Identity is part of the Teleport Infrastructure Identity
```
Is the first sentence here necessary? _Every_ product we list here is part of the platform, so why do we need to say this part for MWI only?

zmb3 commented on docs/pages/teleport-editions.mdx:
```diff
@@ -0,0 +1,115 @@
+
+||Teleport Community Edition|Teleport Enterprise (Self-Hosted)|Teleport Enterprise (Cloud)|
+|---|---|---|---|
+|[Machine ID](./enroll-resources/machine-id/getting-started.mdx)|✖|✔|✔|
+|[Workload Identity](./enroll-resources/workload-identity/getting-started.mdx)|✖|✔|✔|
```
Are we going to remove this stuff from community edition? Machine ID has been in CE since inception.

cc @strideynet

ptgott committed:
Respond to zmb3 feedback

Add minor wording tweaks.

ptgott committed:
Align cells across feature matrix tables

Make it easier for a user to follow a single column across multiple
tables of the feature matrix. Since the style sets cells to 25% width,
which assumes a table that is four cells wide, use a `style` tag so the
style only applies to the tables in the `teleport-editions.mdx` guide.

Closes gravitational/docs-website#134

ptgott committed:
Update the MWI section of the feature matrix

Indicate that it is available for Community Edition users.

ptgott committed:
Edit index.mdx

Apply changes made to identical language in teleport-editions.mdx.

zmb3 commented on docs/pages/teleport-editions.mdx:
```diff
@@ -0,0 +1,126 @@
+Access uses short-lived credentials to enforcing granular access policies and
+Zero-Trust principles.
+
+Teleport Zero Trust Access is available in all Teleport editions, with some
+features only available in Teleport Enterprise.
```
```suggestion
Teleport Zero Trust Access is available in all Teleport editions, with some
features only available in Teleport Enterprise, and some features requiring
self-hosted deployments.
```

Perhaps?

ptgott committed:
Respond to eglenn-teleport feedback

- Move the Community Edition column to the final place, rather than the
  first, since we now want to de-emphasize Community Edition and elevate
  the commercial offering.
- Edit the text to de-emphemphasize "editions", moving the description
  of editions to the end of the page and linking to this from the
  heading of relevant feature matrix columns.
- Remove section intro paragraphs, since the same information is
  on the home page of the docs.

ptgott commented:
@mmcallister Just checking if you have any thoughts on this. Thanks!

benarent commented on docs/pages/feature-matrix.mdx:
```diff
@@ -0,0 +1,95 @@
+|---|:---:|:---:|:---:|
+|[Machine ID](./enroll-resources/machine-id/getting-started.mdx)|✔|✔|✔|
+|[Workload Identity](./enroll-resources/workload-identity/getting-started.mdx)|✔|✔|✔|
+
```
```suggestion
## Teleport Identity Security

||[Teleport Enterprise (Cloud)](#teleport-editions)|[Teleport Enterprise (Self-Hosted)](#teleport-editions)|[Teleport Community Edition](#teleport-editions)|
|---|:---:|:---:|:---:|
|[Identity Security](./admin-guides/teleport-policy/teleport-policy.mdx)|✔|✔|✖|
|[Crown Jewel Monitoring](./admin-guides/teleport-policy/crown-jewels.mdx)|✔|✔|✖|
|[SSH Key Scanning](./admin-guides/teleport-policy/ssh-keys-scan.mdx)|✔|✔|✖|
```

ptgott committed:
Add a Teleport Identity Security section

Co-authored-by: Ben Arent <ben@goteleport.com>

thedevelopnik commented on docs/pages/feature-matrix.mdx:
```diff
@@ -0,0 +1,101 @@
+|[JIT Access Requests](./admin-guides/access-controls/guides/dual-authz.mdx)|✔|✔|Limited|
+|[Session & Identity Locks](./admin-guides/access-controls/guides/locking.mdx)|✔|✔|✖|
+
+## Teleport Machine & Workload Identity
```
@strideynet and I are trying to move away from calling the underlying feature sets in Machine & Workload Identity "Machine ID" and "Workload Identity" as it's both redundant and has created confusion when presented that way, especially since Machine ID was a historic product, and it makes it sound like two distinct products still.

We're now leaning towards MWI Zero Trust Access or MWI Machine Access, and MWI Flexible Workload Identities. Those have tested well in research calls.

I've started putting together for Eddie a [list of features](https://docs.google.com/spreadsheets/d/1Tnx7OBoy9e1MEBLebTBOdRuldplPzcx4kLKEeeyG8eU/edit?usp=sharing) that both overlap those or are specific to one, and which are CE and which are Enterprise-specific (very few).

I'm not super happy with the structure of it right now and invite comment on the sheet. A lot of the ZTA features are simply inherited from the Teleport ZTA features since it's the same logging/locks/etc.

I'd say to get an initial version of this matrix out, we could just change Machine ID and Workload Identity to "Machine Access" and "Flexible Workload Identities." We can then get the more comprehensive matrix out later.

ptgott committed:
Rename MWI features
